### ISSUE TITLE
Getting all collections

### ISSUE THREAD

**Issue by dprice** [nightroman/Mdbc#55]
I'm sure I'm missing something obvious, but I'm just writing a very simple script to query mongo for all databases and a list of all collections and their counts. This is what I'm trying but it's not right:

```
# Load the module
Import-Module Mdbc

$connectionString = 'myConnectionUrl'

$dbs = Connect-Mdbc -ConnectionString $connectionString -DatabaseName *

foreach ($db in $dbs) {
    Write-Host "DB: $db"
    $dbTemp = Get-MdbcDatabase $db
    $collections = Get-MdbcCollection -Name * -Database $dbTemp
    foreach ($collection in $collections) {
        $count = Get-MdbcData -Collection $collection -Count
        Write-Host "- $collection : $count"
    }
}
```

What am I missing here?

**Comment by nightroman** [comment]
(1) Instead of using `-Name *` in `Get-MdbcCollection` you should omit the parameter (or pass null). From help:

> -Name
Specifies the collection name. If it is omitted then all collections are returned.

So this corrected code works (I also replaced `$collection` in output with its more useful name):

```powershell
foreach ($db in $dbs) {
    Write-Host "DB: $db"
    $dbTemp = Get-MdbcDatabase $db
    $collections = Get-MdbcCollection -Database $dbTemp
    foreach ($collection in $collections) {
        $count = Get-MdbcData -Collection $collection -Count
        Write-Host "- $($collection.CollectionNamespace.CollectionName) : $count"
    }
}
```

(2) By using `-Name *` you ask for a collection named `*`. I am surprised that it works but it works. So for each database you get one collection named `*` with 0 records.

**Comment by nightroman** [comment]
Here is the proof that a collection `*` is a thing. This code works and creates the collection `*` with one document:

```powershell
Import-Module Mdbc
Connect-Mdbc . test

$collection = Get-MdbcCollection -Name * -NewCollection
$collection.CollectionNamespace.CollectionName

@{_id = 1; name = 'foo'} | Add-MdbcData
Get-MdbcData
```

**Comment by nightroman** [comment]
Finally, let's take a look at MongoDB manuals https://docs.mongodb.com/manual/reference/limits/

> Collection names should begin with an underscore or a letter character, and cannot:
contain the $.
be an empty string (e.g. "").
contain the null character.
begin with the system. prefix. (Reserved for internal use.)

We can see that `*` is a valid character.
I think we exhausted the issue and may close.

**Comment by dprice** [comment]
I was so close to the answer! Sadly, I did look in the help, but I focused more on the examples rather than the actual parameter help. That was my fault.

As far as the `*` issue, I'm not sure either. I used a `*` based on an example that I think I found scouring through the source code. I can't imagine anyone having a collection named like that, but I am not a hard-core user of Mongo Db. Alternatively, I can't imagine having a table in SQL Server named that. It _seems_ like a "bad thing" ™ to encourage, though.

Thanks again for the help! I have a feeling this module is going to save me a ton of time in the coming year, so thanks in advance!

**Comment by dprice** [comment]
I was so close to the answer! Sadly, I did look in the help, but I focused more on the examples rather than the actual parameter help. That was my fault.

As far as the `*` issue, I'm not sure either. I used a `*` based on an example that I think I found scouring through the source code. I can't imagine anyone having a collection named like that, but I am not a hard-core user of Mongo Db. Alternatively, I can't imagine having a table in SQL Server named that. It _seems_ like a "bad thing to encourage, though.

Thanks again for the help! I have a feeling this module is going to save me a ton of time in the coming year, so thanks in advance!
